polyfill/lib/ecmascript.mjs

```diff
@@ -314,6 +314,8 @@ export const ES = ObjectAssign({}, ES2020, {
     };
   },
   ParseTemporalInstantString: (isoString) => {
+    const match = PARSE.instant.exec(isoString);
+    if (!match) throw new RangeError(`invalid ISO 8601 string: ${isoString}`);
```

Comment:
It might be easier to check after ParseISODateTime that `result.calendar` is undefined? That way there would be a better error message and additionally we wouldn't have to define yet another complicated regex.

Reply:
Is this actually a polyfill bug not a spec bug?  See https://github.com/tc39/proposal-temporal/issues/2143#issuecomment-1100295870